source/core/slang-performance-profiler.cpp

```diff
@@ -39,11 +40,53 @@ namespace Slang
             for (auto func : data)
             {
                 out << func.key << ": \t";
-                out << func.value.invocationCount << "\t" << func.value.duration.count() / 1000000.0f << "\n";
+                auto milliseconds = std::chrono::duration_cast< std::chrono::milliseconds >(func.value.duration);
+                out << func.value.invocationCount << "\t" << milliseconds.count() << "ms\n";
             }
         }
     };
 
+    class SerialPerformaceProfilerImpl : public SerialPerformaceProfiler
+    {
+    public:
+       void accumulateResults(StringBuilder& out, PerformanceProfiler* profile)
+       {
+            PerformanceProfilerImpl* profilerImpl = static_cast<PerformanceProfilerImpl*>(profile);
+            const std::lock_guard<std::mutex> scopeLock(m_mutex);
+
+            for (auto func : profilerImpl->data)
+            {
+                auto milliseconds = std::chrono::duration_cast< std::chrono::milliseconds >(func.value.duration);
+                long ms = milliseconds.count();
+
+                auto entry = m_compileTimeProfiler.find(func.key);
+                if (entry == m_compileTimeProfiler.end())
+                {
+                    m_compileTimeProfiler.emplace(std::make_pair(func.key, 0l));
+                    entry = m_compileTimeProfiler.find(func.key);
+                }
+                entry->second += ms;
+            }
+
+            for (auto entry : m_compileTimeProfiler)
+            {
+                out << entry.first.c_str() << ": \t" << entry.second << " ms\n";
+            }
+       }
+    private:
+        // Those are supposed to accumulate the time spent in each thread
+        std::unordered_map<std::string, long> m_compileTimeProfiler;
+        std::mutex m_mutex;
```

Comment:
Shouldn't this be a static?
As it is, each thread will have their own mutex, isn't it?

Reply:
This class itself is static.

Reply:
I mean the instance of this class is static.

Reply:
Yes, I figured it after leaving my comment.
Thanks for clarifying it tho.